### PR TITLE
Apply delay parameter only to reboot command

### PR DESCRIPTION
**Pull request by sephalon**
Unlike putting a delay in between the reboot command, delaying a on/off
command before execution is not very intuitive; so remove it. See also
issue #72.

### CODE REVIEW

**Comment by cdleonard** [comment]
This means the "delay" argument to on/off is ignored, maybe an exception should be thrown on parsing? Some users might rely on this.

A compatible fix would be to accept and a value of "delay=0"

**Comment by sephalon** [comment]
> This means the "delay" argument to on/off is ignored, maybe an exception should be thrown on parsing? Some users might rely on this.

I agree, but an exception is a quite strong response for that. Maybe we just leave a deprecation message in the log?

> A compatible fix would be to handle value of "delay=0" (this is currently interpreted as unset").

I'm sorry, but I don't quite understand what you mean by that?

**Comment by cdleonard** [comment]
> > This means the "delay" argument to on/off is ignored, maybe an exception should be thrown on parsing? Some users might rely on this.
> 
> I agree, but an exception is a quite strong response for that. Maybe we just leave a deprecation message in the log?

Yes that would be fine. I can't actually think of a valid reason why user would *want* to inject delays for on/off.

> > A compatible fix would be to handle value of "delay=0" (this is currently interpreted as unset").
> 
> I'm sorry, but I don't quite understand what you mean by that?

Right now passing delay=0 is interpreted as delay=5 (I tried this because I also wanted to get rid of delays). I was thinking that a 0 or negative delay could be interpreted to mean "as soon as possible".

**Comment by sephalon** [comment]
> Yes that would be fine. I can't actually think of a valid reason why user would want to inject delays for on/off.

I came up with a more backwards compatible solution now; please review and let me know what you think, then I'll squash the two commits.

> Right now passing delay=0 is interpreted as delay=5 (I tried this because I also wanted to get rid of delays). I was thinking that a 0 or negative delay could be interpreted to mean "as soon as possible".

As per my understanding, `delay=0` always worked as intended, it was just not very intuitive. Can you please check again?

File: pdudaemon/listener.py [hunk 1]
```diff
@@ -67,12 +67,17 @@ def parse_http(data, path):
 
 
 def process_request(args, config, db_queue):
+    if args.request in ["on", "off"] and args.delay:
+        logger.warn("delay parameter is deprecated for on/off commands")
     if args.delay:
```
[the remark this answers is not in the document]
Comment (sephalon) [reply pdudaemon/listener.py]: Correct, fixed.

**Comment by mattface** [comment]
Thanks for the patch, the changes make sense to me so I'm happy to merge

**Pull request merged**